### ISSUE TITLE
first client not able to emit signal(or server can't catch first client signal)

### ISSUE THREAD

**Issue by inyourpc** [socketio/socket.io#1186]
var express = require('express')
  , io = require('socket.io')
  //, io2= require('/lib/socket.io')
  , routes = require('./routes')
  , http = require('http')
  , path = require('path');

var app = express();

app.configure(function(){
  app.set('port', process.env.PORT || 3000);
  app.set('views', __dirname + '/views');
  app.set('view engine', 'ejs');
  app.use(express.favicon());
  app.use(express.logger('dev'));
  app.use(express.bodyParser());
  app.use(express.methodOverride());
  app.use(express.cookieParser('your secret here'));
  app.use(express.session());
  app.use(app.router);
  app.use(require('stylus').middleware(__dirname + '/public'));
  app.use(express.static(path.join(__dirname, 'public')));
});

app.configure('development', function(){
  app.use(express.errorHandler());
});

//app.get('/', routes.index);
app.get('/', function(req, res){
  res.render('index', {
    title: 'Home'
  });
});

app.get('/about', function(req, res){
  res.render('about', {
    title: 'About'
  });
});

app.get('/contact', function(req, res){
  res.render('contact', {
    title: 'Contact'
  });
});
/*
http.createServer(app).listen(app.get('port'), function(){
  console.log("Express server listening on port " + app.get('port'));
});
*/
//Now implementing websocket to listen 
var server = http.createServer(app).listen(app.get('port'), function(){
  console.log("Express server listening on port " + app.get('port'));
});

var Games=new Array();
//creating public folder to cached will it ,let's see.
var oneYear = 31557600000;
app.use(express.static(__dirname + '/public', { maxAge: oneYear }));

//socket implementation 
var sio = io.listen(server);
sio.sockets.on('connection', function (socket) {
function checkPlayers(ServerName,pl_id){

```
//////////////////////////////////////
```

  socket.on('HideCard',function (cardToHide){

  });
/_-----------------------------------INROUND-------------------------------_/  
socket.on('updateTurn',function (card){

```
}
```

}//function ends here.....
...........client side......................................................................................................
<script src='/socket.io/socket.io.js'></script> ...
console.log('sending signal to server,once the card is cicked');
socket.emit('updateTurn',evt.target.name);
-------------------------******************************************------
So basically when the page is served to 4 clients from server ,the first cilent is not able to send the signal i.e emit it to the server ,while other 3 are able to send it to the server....so is there scope issue that i have written socket's code at server inside function ...What are your suggestions ...P.S how do i know that what mechansim is being used by transmitting signal ,can use several fallback as well for more accuracy and surity...
